documentation/smg/16-serialiser.md

```diff
@@ -0,0 +1,275 @@
+# Serialisation details
+
+As of the integration of the C++ serialiser, the serialised format has been updated to be more consistent and to have greater capabilities.
+
+For reference, the old format is also detailed below
+
+## New format
+
+### Standard format
+The new style has a standardised header format depending on the shape of the data (with the exception of function handles).
+This format allows extension to support arrays of objects and structs in a way which the old serialiser couldn't.
+It also greatly simplifies the tag structure and allows it to align with the C++ MEX API data types.
+Data in these cases is serialised in the linearised order as defined by `data(:)` and the array reshaped by the deserialiser.
+
+This standard format for the header is as follows:
+RANK |     tag    | dim1 | dim2 | ... | Data
+-----|------------|------|------|-----|------
+NULL | 32 + tag   | 0    |      |     |
+0    |    tag     |      |      |     | Data
+1    | 32 + tag   | nElem|      |     | Data
+2    | 64 + tag   | dim1 | dim2 |     | Data
+N    | N<<5 + tag | dim1 | dim2 | ... | Data
```

Comment:
In the tag column, presuming (possibly incorrectly) that tag is the tag structure described below and then 32,64,N<<5 are further bits in front, or offsets? Also how is NULL recorded?

Looking again at the tag description below, that contains 3 bits for the rank, and 5 for the tag. So it looks as if rank and tag are not 2 separate items but one combined item. Is that correct? And does 32 mean bit=1 at the 32 position etc. And is N limited to max 7 as only 3 rank bits are available? (noted that this is confirmed in the limitations section).

Finally, the tag header on the column and the tag value in the column cell are a bit inconsistent. The column tag appears to be rank+type tag